pkg/asset/templates/kubelet.yaml

```diff
@@ -17,27 +17,19 @@ spec:
       - name: kubelet
         image: quay.io/coreos/hyperkube:v1.3.0_coreos.1
         command:
-        - /nsenter
-        - --target=1
-        - --mount
-        - --wd=.
-        - --
-        - ./hyperkube
-        - kubelet
-        - --api-servers={{ index .APIServers 0 }}
-        - --config=/etc/kubernetes/manifests
-        - --allow-privileged
-        - --hostname-override=$(MY_POD_IP)
-        - --cluster-dns=10.3.0.10
-        - --cluster-domain=cluster.local
-        - --kubeconfig=/etc/kubernetes/kubeconfig
-        - --lock-file=/var/run/lock/kubelet.lock
-        - --minimum-container-ttl-duration=3m0s
+        - /bin/bash
+        - -c
+        - "/nsenter --target=1 --mount --wd=. -- ./hyperkube kubelet --hostname-override=$(MY_POD_IP) $(FLAGS)"
         env:
           - name: MY_POD_IP
             valueFrom:
               fieldRef:
                 fieldPath: status.podIP
+          - name: FLAGS
+            valueFrom:
+              configMapKeyRef:
+                name: kubelet-config
```

Comment:
So instead of the kubelet DS depending on the configMap directly it will essentially `cat /etc/kubernetes/kubelet.env` for it's flags? That seems reasonable.

What do you think, then, about having the node-agent kick the kubelet pod for it's node? That way, the node agent updates config, and then rolls both the on-host and self-hosted kubelets to get new config.